eogrow/core/schemas.py

```diff
@@ -47,6 +47,13 @@ class PipelineSchema(BaseSchema):
     workers: int = Field(
         1, description="Number of workers for parallel execution of workflows. Parameter does not affect ray clusters."
     )
+    worker_type: Optional[str] = Field(
```

Comment:
I'm just thinking about how this generalizes, or if there is a need to be generalized:
- will we have tasks where we can specify multiple suitable worker types?
- will we have tasks that will _require_ multiple custom resources? e.g. if we ever need to parallelize training over multiple folds on an instance with multiple GPUs, once could create a custom resource of 1 GPU and 1 CPU

Reply:
this doesn't generalize. we chose to keep it simple at this point.
If we ever need more elaborate resource requests, then this field should just be `ray_remote_kwargs` and nothing else. But then you'd have to say `"ray_remote_kwargs" = {"resources": {"BigRamWorker": 1}}` instead of `"worker_type" = "BigRamWorker"} and it's also less intuitive what we're trying to achieve here.

For the first bullet point, workers can have multiple tags. So if a pipeline can be run on multiple workers, just create a custom tag for it and add it to those workers.

Not sure what you mean with the second one

Reply:
another example pops to mid. Say we have a big ram worker `r5.2xlarge` (8 CPUs, 64 GB RAM) and one pipeline requires 25 GB per task, while another requires 50 GB per task.

do we:
- create two different worker types in the YAML (`BigRamWorkerA` (with 32 GB instance) and `BigRamWorkerB` (with 64 GB instance)), where worker node has the amount of CPU resources set to 1 in both cases, or
- could we say for pipeline A that we need `"ray_resources" = {"BigRamWorker": 0.5, "CPU": 4}` and for pipeline B `"ray_resources" = {"BigRamWorker": 1,  "CPU": 8}`

In the second suggestion, (if I didn't mess something up), we could be running both pipelines efficiently and on the same worker (since there would be 2 processes for pipeline A running on the worker at the same time)

Reply:
but, this requires you know how many CPUs are on the worker.

Reply:
there is also an option to request memory as a resource, so that would probably fit better :)

as i said, there was a discussion on which path to choose, and we said we'd go with the 'worker_type' for now. It's simple to change it to full resource management later, but you can also re-open that discussion

Reply:
due to some issue with notifications, I missed the discussion, so i will respect the decisions made there. Going with the first option here is completely valid as well

Reply:
It's really something that we can easily change in the future (it's just how the parameters are passed from the config to the executor) so i wouldn't lose sleep over it.